ibis/expr/scope.py

```diff
@@ -41,19 +41,63 @@
 from ibis.expr.timecontext import TimeContextRelation, compare_timecontext
 from ibis.expr.typing import TimeContext
 
-ScopeItem = namedtuple('item', ['value', 'timecontext'])
+ScopeItem = namedtuple('item', ['timecontext', 'value'])
 
 
 class Scope:
-    def __init__(self, items: Dict[str, ScopeItem] = None):
-        self._items = items or {}
+    def __init__(
+        self,
+        param: Dict[Node, Any] = None,
+        timecontext: Optional[TimeContext] = None,
+    ):
+        """ Take a dict of `op`, `result`, create a new scope and save
+        those pairs in scope. Associate None as timecontext by default.
+        This is mostly used to init a scope with a set of given params.
+        """
+        self._items = (
+            {op: ScopeItem(timecontext, value) for op, value in param.items()}
+            if param
+            else {}
+        )
 
     def __contains__(self, op):
+        """ Given an `op`, return if `op` is present in Scope.
```

Comment:
I think it is still useful when we want to iterate items in scope. 

Things like:
```
If op in scope:
    return scope[op]
```
should all be replaced because it is trying to get value from scope. But if we want to do something else, like adjust all contexts in scope, or get all values from scope where we don't care about time contexts, or get max time context in current scope, (some cases that I could think of) we may need `__contains__` for its simplicity.